### PR TITLE
Pcm 7616 Fixing all Email Notification and RH Watcher issues

### PR DESCRIPTION
**Pull request by vrathee**
@gandhikeyur @praniredhat @anujsi Please review

### CODE REVIEW

File: app/cases/controllers/new.js [hunk 1]
```diff
@@ -167,13 +167,15 @@ export default class New {
...
+        const setUsers = () => {
             if(securityService.loginStatus.authedUser.is_internal || securityService.loginStatus.authedUser.org_admin){
                 $scope.usersOnAccount = _.cloneDeep(CaseService.users);
                 $scope.usersOnAccount = $scope.usersOnAccount.concat(securityService.loginStatus.authedUser.accountContacts);
                 $scope.usersOnAccount = _.uniqBy($scope.usersOnAccount,'sso_username');
             }
-        });
+        }
+        $scope.$watch('CaseService.users', () => setUsers());
+        $scope.$watch('securityService.loginStatus.authedUser.accountContacts', () => setUsers());
```
Comment (vrathee) [inline app/cases/controllers/new.js]: Was only adding securityService.loginStatus.authedUser.accountContact if they have been loaded before CaseService.users else they were not part of the options and selection

File: app/cases/controllers/emailNotifySelect.js [hunk 2]
```diff
@@ -84,30 +56,36 @@ export default class EmailNotifySelect {
...
+                    $scope.usersOnAccount = _.cloneDeep(CaseService.users);
                 }
+                $scope.selectedUsers =  _.intersection(_.map($scope.usersOnAccount, 'sso_username'), CaseService.originalNotifiedUsers);
+                $scope.selectedUsers.unshift(CaseService.kase.contact_sso_username); // fake insert the Case Contact
             }
+            CaseService.populateUsers();
+            $scope.$watch('CaseService.users', () => setUsers());
+            $scope.$watch('CaseService.originalNotifiedUsers', () => setUsers());
+            $scope.$watch('securityService.loginStatus.authedUser.accountContacts', () => setUsers());
+            securityService.loginStatus.authedUser.is_internal && $scope.$watch('CaseService.internalNotificationContacts', () => setUsers());
```
Comment (vrathee) [inline app/cases/controllers/emailNotifySelect.js]: Making sure we populate the dropdown with all the required users.

File: app/cases/views/emailNotifySelect.jade [hunk 3]
```diff
@@ -1,22 +1,15 @@
...
-                ng-click="toggleCurrentUser()",
-                ng-disabled="saving || isCurrentUserCaseContact()",
-                ng-show="internal && !CaseService.redhatUsersLoading")
-                span(ng-hide="isCurrentUserWatcher()", translate) Add me as watcher
-                span(ng-show="isCurrentUserWatcher()", translate) Remove me as watcher
             button.btn.btn-info.btn-xs(
                 ng-click="toggleCurrentUser()",
                 ng-disabled="saving || isCurrentUserCaseContact()",
                 ng-show="!internal && !CaseService.usersLoading && !securityService.loginStatus.authedUser.org_admin && !securityService.loginStatus.authedUser.is_internal")
                     span(ng-hide="isCurrentUserNotifiedUser() || isCurrentUserCaseContact()", translate) Add me as watcher
```
Comment (praniredhat) [inline app/cases/views/emailNotifySelect.jade]: @vrathee  ```!securityService.loginStatus.authedUser.org_admin && !securityService.loginStatus.authedUser.is_internal```  Is not operator intendted here ?

File: app/cases/controllers/emailNotifySelectInternal.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+import _ from 'lodash';
+import hydrajs  from '../../shared/hydrajs';
+
+export default class EmailNotifySelectInternal {
+    constructor($scope, CaseService, securityService, AlertService, strataService, CASE_EVENTS, $filter, RHAUtils, EDIT_CASE_CONFIG, gettextCatalog) {
+        'ngInject';
+        $scope.isLoadingContacts = false;
+        $scope.CaseService = CaseService;
+        $scope.securityService = securityService;
```
Comment (praniredhat) [inline app/cases/controllers/emailNotifySelectInternal.js]: @vrathee Should we keep securityService in the scope, though we don't have any usages in the jade.  And also caseService, we only use one field from caseService?

**Comment by praniredhat** [comment]
Reviewed.

File: app/cases/views/emailNotifySelect.jade (hunk 3 above)
Comment (vrathee) [reply app/cases/views/emailNotifySelect.jade]: @praniredhat Not is intentional here, I removed !internal as it was not needed.
Regarding using ng-hide ,  we will need to reverse the condition to do so.. so its ok

**Pull request merged**